docs/DeveloperGuide.md

```diff
@@ -1,34 +1,509 @@
+</div>
+**Step 2** : `execute()` is called from the instance of `AddModuleCommand`. It can throw `AcademicException` 
+or `IOException`. `FileHandler` and `Logger` classes from the _java API_ are instantiated to handle logging for the remainder of the `execute()` method. 
+
+**Step 3** : `in` then reads in the next two lines of input, which is the user's input for the desired semester for the `moduleCode` and `moduleCode`'s grades.
+
+**Step 4** : `validateInputs()` is called to validate the user entered data against `allModules`.
+
+**Step 5** : `AddUtils` is called upon to return module credit for `moduleCode` by `getModuleCreditForModule()`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_state6.png" alt="State diagram for AddModuleCommand Step 6"/>
+</div>
+
+**Step 6** :  `AddUtils` is called upon again to add the module's data to the user, by instantiating a new
+`PartialModule` and storing it in both `userModuleList` and `userModuleMap` via `Person`.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_finalState.png" alt="Final state diagram for AddModuleCommand"/>
+</div>
+
+**Step 7** : `FileHandler`, `Logger`, `PartialModule`, `ModuleValidator`, `AddUtils` and `AddModuleCommand` are terminated.
+
+The following sequence diagram shows how the `AddModuleCommand` works:
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_sequence.png" alt="Sequence diagram for AddModuleCommand"/>
+</div>
+
+The following activity diagram summarizes what happens when the user executes a `AddModuleCommand` :
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/addModuleCommand_activity.png" alt="Activity diagram for AddModuleCommand"/>
+</div>
+
+#### Design consideration
+
+The following options were considered when implementing commands:
+
+* Option 1 (Current Implementation): Implementing each command as a class by itself
+    * Pros: Increases modularity of code, higher overall code quality 
+    * Cons: More complicated to implement
+* Option 2: Implementing each command as a method in a class
+    * Pros: Easier to implement
+    * Cons: Class needs to be instantiated and increases coupling, reducing testability. This method also decreases SLAP.
+
+### Academic Calendar Planner: Edit Module Feature
+
+#### Current implementation
+
+Similar to the add module command, the edit module command is also executed by `AcademicPlannerParser`. It allows the user to edit the existing modules added to their `Academic Planner` by accessing the specified `PartialModule` object within the `userModuleList`and `userModuleMap`. 
+
+Given below is an example usage scenario and how add module command behaves at each step.
+
+<div style="text-align:center">
+    <img src="./images/DeveloperGuide/editModuleCommand_initialState.png" alt="Initial state diagram for Edit Module Command"/>
+</div>
```

Comment:
Maybe change the boxes, as the rounded boxes are against standard uml notation
![image](https://user-images.githubusercontent.com/45010376/97514588-1157ff80-19ca-11eb-80b8-6160a9c2f333.png)